src/components/Checkbox/Checkbox.tsx

```diff
@@ -39,67 +50,95 @@ export const onChangeDefault = () => {
   return;
 };
 
+function CheckboxIcon(props) {
+  const { isIndeterminate, isChecked, ...rest } = props;
+
+  const d = isIndeterminate
+    ? "M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm0,19a1.5,1.5,0,1,1,1.5-1.5A1.5,1.5,0,0,1,12,19Zm1.6-6.08a1,1,0,0,0-.6.917,1,1,0,1,1-2,0,3,3,0,0,1,1.8-2.75A2,2,0,1,0,10,9.255a1,1,0,1,1-2,0,4,4,0,1,1,5.6,3.666Z"
+    : "M8.795 15.875l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41-10.59 10.58z";
+
+  /*
+    boxSize - controls the size of the "checked" icon.
+    color - does absolutely nothing?
+    path:fill - can only control color with hardcoded value?
+  */
+  return (
+    <Icon viewBox="0 0 24 24" boxSize={6} color="ui.focus" {...rest}>
+      <path fill="#4181f1" d={d} />
+    </Icon>
+  );
+}
+
 const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
   (props, ref?) => {
     const {
-      checked,
       className,
-      disabled,
-      errored,
       errorText,
       helperText,
       id,
+      isChecked,
+      isDisabled = false,
+      isInvalid = false,
+      isRequired = false,
       labelText,
       name,
-      showLabel,
+      showLabel = true,
       value,
     } = props;
 
-    const attributes = props.attributes || {};
-    const modifiers = props.modifiers ? props.modifiers : [];
+    const styles = useMultiStyleConfig("Checkbox", {});
+    const footnote = isInvalid ? errorText : helperText;
+    const attributes = {};
     const checkboxID = id || generateUUID();
     const onChange = props.onChange || onChangeDefault;
 
-    const footnote = errored ? errorText : helperText;
-
     if (!showLabel) {
       attributes["aria-label"] =
         labelText && footnote ? `${labelText} - ${footnote}` : labelText;
     } else {
       if (footnote) attributes["aria-describedby"] = `${checkboxID}-helperText`;
     }
 
-    if (errored) {
-      modifiers.push("error");
-    }
-
     return (
       <>
-        <input
-          {...(checked !== undefined
+        <ChakraCheckbox
+          id={checkboxID}
+          className={className}
+          name={name || "default"}
+          {...(isDisabled && {
+            isDisabled: true,
+          })}
+          isInvalid={isInvalid}
+          isRequired={isRequired}
+          ref={ref}
+          value={value}
+          {...(isChecked !== undefined
             ? {
-                checked: checked,
+                isChecked: isChecked,
                 onChange: onChange,
               }
             : {
-                defaultChecked: false,
+                defaultIsChecked: false,
               })}
-          className={bem("checkbox", modifiers, "input", [className])}
-          disabled={disabled}
-          id={checkboxID}
-          name={name || "default"}
-          ref={ref}
-          type="checkbox"
-          value={value}
+          size="lg"
+          // @TODO why not just set these in the checkbox theme?
+          colorScheme="white"
+          borderColor="ui.black"
```

Comment:
Agreed, and `size` and `colorScheme` are already a part of the `defaultProps` object in `Checkbox.ts` so you can remove `size`, `colorScheme`, and `borderColor` here.